src/components/App/App.tsx

```diff
@@ -4,109 +4,57 @@ import { getData } from '../../utils/apiCalls';
-  markets: {
-    id: number;
-    distanceFromZip: number;
-    marketName: string;
-    schedule: {
-      dayOfWeek: string;
-      time: string;
-      season: string;
-    }[];
-  }[];
-  marketDetails: {
-    id: number;
-    street: string;
-    city: string;
-    state: string;
-    zip: string;
-    schedule: {
-      dayOfWeek: string;
-      time: string;
-      season: string;
-    }[];
-    products: string[];
-    mapsLink: string;
-    marketName: string;
-  }[];
-  //zip: string;
-  errorCode?: string;
-}
-
-interface OneDetail {
-  oneDetail: {
-    id: number;
-    street: string;
-    city: string;
-    state: string;
-    zip: string;
-    schedule: {
-      dayOfWeek: string;
-      time: string;
-      season: string;
-    }[];
-    products: string[];
-    mapsLink: string;
-    marketName: string;
-  };
-  zip: string;
-  errorCode?: string;
-}
+import ScrollToTop from '../../utils/scrollToTop';
+import { ApiMarkets, Market } from '../../types';
+import loadingIcon from '../../images/loading.jpg';
 
 export const App: React.FC = () => {
   const [allMarkets, setMarkets] = useState<ApiMarkets['markets']>([]);
   const [marketDetails, setDetails] = useState<ApiMarkets['marketDetails']>([]);
-  const [selectedMarket, setSelectedMarket] =
-    useState<OneDetail['oneDetail']>();
+  const [selectedMarket, setSelectedMarket] = useState<Market>();
   const [zip, setZip] = useState<string>('');
-  const [errorCode, setErrorCode] = useState<ApiMarkets["errorCode"]>("");
-  const [ loading, setLoading ] = useState<boolean>(true);
+  const [errorCode, setErrorCode] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
   const history = useHistory();
 
   const getMarkets = async (zip: string, distance: number) => {
+    setErrorCode('');
+    setLoading(true);
     setZip(zip);
     try {
-      setLoading(true)
-      let response = await getData(`zipSearch?zip=${zip}`);
-      let data = await checkForError(response);
+      let data = await getData(`zipSearch?zip=${zip}`);
       let cleanedData = cleanMarketsData(data.results, distance);
-      getDetails(cleanedData);
+      await getDetails(cleanedData);
       history.push('/markets');
     } catch (error) {
       setErrorCode(error.message);
-      history.push('/markets');
     }
   };
 
-  const getDetails = (filteredMarkets: ApiMarkets['markets']) => {
-    Promise.all(
-      filteredMarkets.map(async currentMarket => {
-        const response = await getData(`mktDetail?id=${currentMarket.id}`);
-        const data = await checkForError(response);
-        return cleanDetailsData(data.marketdetails, currentMarket.id);
-      })
-    )
-    .then(data => addScheduleToMarkets(data, filteredMarkets))
-    .then(completeData => setData(completeData))
+  const getDetails = async (filteredMarkets: ApiMarkets['markets']) => {
+    try {
+      let data = await Promise.all(
+        filteredMarkets.map(async currentMarket => {
```

Comment:
I _think_ but might be wrong that async is only needed in the function declaration.

Reply:
I just did what VSC told me to do to convert to async haha